cmd/upgrade/accountroles/accountroles.go

```diff
@@ -0,0 +1,160 @@
+	Short:   "Upgrade account-wide IAM roles to the latest version.",
+	Long:    "Upgrade account-wide IAM roles to the latest version before upgrading your cluster.",
+	Example: `  # Upgrade account roles for ROSA STS clusters
+  rosa upgrade account-roles`,
+	Run: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVar(
+		&args.prefix,
+		"prefix",
+		aws.DefaultPrefix,
+		"User-defined prefix for all generated AWS resources",
+	)
+	Cmd.MarkFlagRequired("prefix")
+	confirm.AddFlag(flags)
+	interactive.AddFlag(flags)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	prefix := args.prefix
+
+	// Create the AWS client:
+	awsClient, err := aws.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create AWS client: %v", err)
+		os.Exit(1)
+	}
+
+	// Create the client for the OCM API:
+	ocmClient, err := ocm.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	creator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Failed to get IAM credentials: %s", err)
+		os.Exit(1)
+	}
+
+	isUpgradeNeed, err := awsClient.IsUpgradedNeededForRole(prefix, creator.AccountID)
+	if err != nil {
+		reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+	if !isUpgradeNeed {
+		reporter.Infof("Account role with the prefix '%s' is already up-to-date.", prefix)
+		os.Exit(1)
+	}
+	reporter.Infof("Starting to upgrade the policies")
+
+	err = upgradeAccountRolePolicies(reporter, awsClient, prefix, creator.AccountID)
+	if err != nil {
+		reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+}
+
+func upgradeAccountRolePolicies(reporter *rprtr.Object, awsClient aws.Client, prefix string, accountID string) error {
+	for file, role := range aws.AccountRoles {
+		name := sts.GetRoleName(prefix, role.Name)
+		if !confirm.Prompt(true, "Upgrade the '%s' role polices to version %s?", name,
+			aws.DefaultPolicyVersion) {
+			continue
+		}
+		filename := fmt.Sprintf("sts_%s_permission_policy.json", file)
+		path := fmt.Sprintf("templates/policies/%s", filename)
+		policyARN := sts.GetPolicyARN(accountID, fmt.Sprintf("%s-Policy", name))
+
+		policy, err := aws.ReadPolicyDocument(path)
+		if err != nil {
+			return err
+		}
+		reporter.Debugf("Creating permission policy '%s'", policyARN)
+		policyARN, err = awsClient.EnsurePolicy(policyARN, string(policy),
+			aws.DefaultPolicyVersion, map[string]string{
+				tags.OpenShiftVersion: aws.DefaultPolicyVersion,
+				tags.RolePrefix:       prefix,
+				tags.RoleType:         file,
+			})
+		if err != nil {
+			return err
+		}
+		reporter.Debugf("Attaching permission policy to role '%s'", filename)
+		err = awsClient.AttachRolePolicy(name, policyARN)
```

Comment:
Since we're creating a new version of the existing policy, do we even need to attach it again?

Reply:
tested and we dont need attach again